modules/biobank/php/containercontroller.class.inc

```diff
@@ -0,0 +1,405 @@
+        $containerList = array();
+        foreach ($list as $i => $item) {
+            $container = $this->dao->getInstanceFromArray($item);
+            $this->validateInstance($container);
+            $containerList[$i] = $container;
+        }
+
+        // Save Containers
+        $containerIds = [];
+        foreach ($containerList as $i => $container) {
+            $containerIds[$i] = $this->dao->saveInstance($container);
+        }
+
+        return $containerIds;
+    }
+
+    // TODO: Decide if this function should take a list of containers, rather
+    // than a singular container.
+    /**
+     * This method takes in a container array from the front-end,
+     * validates them, instantiates them and then sends them to the DAO to be
+     * saved to the database.
+     *
+     * @param array $containerArray container array to be created
+     *
+     * @return void
+     */
+    public function updateInstance(array $containerArray) : void
+    {
+        $this->validatePermission('PUT');
+        $container = $this->dao->getInstanceFromArray($containerArray);
+        $this->validateInstance($container);
+        $this->dao->saveInstance($container);
+    }
+
+    /**
+     * Get all data associated with the Container Entity. This will be used
+     * for front-end mapping and populating form options.
+     *
+     * @return array Associative array of option data.
+     */
+    public function getOptions() : array
+    {
+        $this->validatePermission('GET');
+        return array(
+                'types'           => $this->dao->getTypes(),
+                'typesPrimary'    => $this->dao->getTypes(['Primary' => 1]),
+                'typesNonPrimary' => $this->dao->getTypes(['Primary' => 0]),
+                'dimensions'      => $this->dao->getDimensions(),
+                'coordinates'     => $this->dao->getCoordinates(),
+                'stati'           => $this->dao->getStati(),
+               );
+    }
+
+    /**
+     * Treats the Container DAO as a Provisioner that can be iterated through
+     * to provide the permissable Container Objects for the current User.
+     *
+     * @return \LORIS\Data\Provisioner
+     */
+    private function getDataProvisioner() : \LORIS\Data\Provisioner
+    {
+        $dao = new ContainerDAO($this->db);
+        if ($this->user->hasPermission('access_all_profiles') === false) {
+            $dao = $dao->filter(new \LORIS\Data\Filters\UserSiteMatch());
+        }
+        if ($this->user->hasPermission('biobank_container_view') === false) {
+            $dao = $dao->filter(new PrimaryContainerFilter(0));
+        }
+        if ($this->user->hasPermission('biobank_specimen_view') === false) {
+            $dao = $dao->filter(new PrimaryContainerFilter(1));
+        }
+        return $dao;
+    }
+
+    /**
+     * Checks User Permissions for creating or updating Container Objects.
+     *
+     * @param string $action User requested action.
```

Comment:
```suggestion
     * @param string $method User requested method.
```
Technically these are HTTP request methods so this variable name would be a bit better.